war/src/main/js/components/command-palette/index.js

```diff
@@ -0,0 +1,160 @@
+import { LinkResult } from "@/components/command-palette/models";
+import { JenkinsSearchSource } from "./datasources";
+import debounce from "lodash/debounce";
+import * as Symbols from "./symbols";
+import makeKeyboardNavigable from "@/util/keyboard";
+
+const datasources = [JenkinsSearchSource];
+
+function init() {
+  const i18n = document.getElementById("command-palette-i18n");
+  const headerCommandPaletteButton = document.getElementById(
+    "button-open-command-palette"
+  );
+  const commandPalette = document.getElementById("command-palette");
+  const commandPaletteWrapper = commandPalette.querySelector(
+    ".jenkins-command-palette__wrapper"
+  );
+  const commandPaletteInput = document.getElementById("command-bar");
+  const commandPaletteSearchBarContainer = commandPalette.querySelector(
+    ".jenkins-command-palette__search"
+  );
+  const searchResults = document.getElementById("search-results");
+  const searchResultsContainer = document.getElementById(
+    "search-results-container"
+  );
+
+  const hoverClass = "jenkins-command-palette__results__item--hover";
+
+  makeKeyboardNavigable(
+    searchResultsContainer,
+    () => searchResults.querySelectorAll("a"),
+    hoverClass
+  );
+
+  // Events
+  headerCommandPaletteButton.addEventListener("click", async function () {
+    if (commandPalette.hasAttribute("open")) {
+      hideCommandPalette();
+    } else {
+      await showCommandPalette();
+    }
+  });
+
+  commandPaletteWrapper.addEventListener("click", function (e) {
+    if (e.target !== e.currentTarget) {
+      return;
+    }
+
+    hideCommandPalette();
+  });
+
+  async function renderResults() {
+    const query = commandPaletteInput.value;
+    let results;
+
+    if (query.length === 0) {
+      results = [
+        new LinkResult(
+          Symbols.HELP,
+          i18n.dataset.getHelp,
+          "https://www.jenkins.io/redirect/search-box",
+          true,
+          document
+            .getElementById("page-header")
+            .dataset.searchHelpUrl.escapeHTML(),
+          true
+        ),
+      ];
+    } else {
+      await Promise.all(datasources.map((ds) => ds.execute(query))).then(
+        (response) => {
+          results = response.flat();
+        }
+      );
+    }
+
+    // Clear current search results
+    searchResults.innerHTML = "";
+
+    if (query.length === 0 || Object.keys(results).length > 0) {
+      results.forEach(function (obj) {
+        const renderedObject = obj.render();
+
+        let link = document.createElement("DIV");
+        link.innerHTML = renderedObject;
```

Comment:
❌ 
Haven't dug to find which part of the code is the source for this (multiple innerHTML) but there is an XSS around there.

Reproduction steps:
1) create a job with any name
2) set its display name to something like `blabla <img src=x onerror=alert(123) /> blabla`
3) search for "b", it will trigger the alert